Source/LuaMachine/Private/LuaStateComponent.cpp

```diff
@@ -0,0 +1,39 @@
+// Fill out your copyright notice in the Description page of Project Settings.
+
+
+#include "LuaStateComponent.h"
+
+
+ULuaStateComponent::ULuaStateComponent()
+{
+	PrimaryComponentTick.bCanEverTick = true;
+}
+
+ULuaState* ULuaStateComponent::GetLuaState()
+{
+	if (LuaState)
+	{
+		return LuaState;
+	}
+
+	if (!LuaStateClass)
+	{
+		return nullptr;
+	}
+
+	LuaState = NewObject<ULuaState>(this, LuaStateClass);
+	LuaState->GetLuaState(GetWorld());
+	LuaState->ReceiveLuaLevelAddedToWorld(GetOwner()->GetLevel(), GetWorld());
```

Comment:
I _believe_ here we're going to want to add:

```
LuaState->AddToRoot();
```

With a corresponding ->RemoveFromRoot(); on a destructor/DestroyLuaState().

Without this the LuaState will get garbage collected out from under you if you don't reference it anywhere in blueprints. 

Ran into this issue as we were only accessing it via the pure getters.

Reply:
I recommend, just adding the UPROPERTY macro above the lua state reference, this should keep it protected from garbage collection as long as the outer is the LuaStateComponent